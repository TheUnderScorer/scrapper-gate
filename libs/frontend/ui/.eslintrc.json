{
  "extends": ["plugin:@nrwl/nx/react", "../../../.eslintrc.json"],
  "ignorePatterns": ["!**/*"],
  "overrides": [
    {
      "files": ["*.ts", "*.tsx", "*.js", "*.jsx"],
      "parserOptions": {
        "project": [
          "libs/frontend/ui/tsconfig.*?.json",
          "libs/frontend/ui/.storybook/tsconfig.json"
        ]
      },
      "rules": {}
    },
    {
      "files": ["*.ts", "*.tsx"],
      "rules": {}
    },
    {
      "files": ["*.js", "*.jsx"],
      "rules": {}
    }
  ]
}
