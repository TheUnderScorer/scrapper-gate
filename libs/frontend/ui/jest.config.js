module.exports = {
  displayName: 'frontend-ui',
  preset: '../../../jest.preset.js',
  transform: {
    '^.+\\.[tj]sx?$': 'ts-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
  coverageDirectory: '../../../coverage/libs/frontend/ui',
  setupFilesAfterEnv: [`${process.cwd()}/tests/setupFrontendTests.ts`],
};
