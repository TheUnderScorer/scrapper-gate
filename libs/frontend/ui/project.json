{
  "root": "libs/frontend/ui",
  "sourceRoot": "libs/frontend/ui/src",
  "projectType": "library",
  "targets": {
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "options": {
        "lintFilePatterns": ["libs/frontend/ui/**/*.{ts,tsx,js,jsx}"]
      },
      "outputs": ["{options.outputFile}"]
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["coverage/libs/frontend/ui"],
      "options": {
        "jestConfig": "libs/frontend/ui/jest.config.js",
        "passWithNoTests": true
      }
    },
    "storybook": {
      "executor": "@nrwl/storybook:storybook",
      "options": {
        "uiFramework": "@storybook/react",
        "port": 4400,
        "config": {
          "configFolder": "libs/frontend/ui/.storybook"
        }
      },
      "configurations": {
        "ci": {
          "quiet": true
        }
      }
    },
    "build-storybook": {
      "executor": "@nrwl/storybook:build",
      "outputs": ["{options.outputPath}"],
      "options": {
        "uiFramework": "@storybook/react",
        "outputPath": "dist/storybook/frontend-ui",
        "config": {
          "configFolder": "libs/frontend/ui/.storybook"
        }
      },
      "configurations": {
        "ci": {
          "quiet": true
        }
      }
    }
  },
  "tags": []
}
