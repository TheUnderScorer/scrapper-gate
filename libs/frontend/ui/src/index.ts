export * from './atoms/Centered/Centered';
export * from './atoms/Emoji/Emoji';
export * from './molecules/ErrorAlert/ErrorAlert';
export * from './molecules/SimpleDialog/SimpleDialog';
export * from './molecules/Scoped/Scoped';
export * from './styles/styles.types';
export * from './styles/highlight';
export * from './styles/Styles';
export * from './molecules/Layout/Layout';
export * from './organisms/ControlledList/ControlledList';
export * from './organisms/FlowBuilder';
export * from './assets/getAsset';
export * from './assets/useAsset';
export * from './assets/assets.types';
export * from './atoms/Image/Image';
export * from './molecules/QueryDrawer/QueryDrawer';
export * from './molecules/Dropdown/Dropdown';
export * from './atoms/TooltipText/TooltipText';
export * from './molecules/TextWithKeyHint/TextWithKeyHint';
export * from './molecules/UndoButtons/UndoButtons';
export * from './atoms/Buttons/Buttons';
export * from './molecules/ResizablePanel/ResizablePanel';
export * from './molecules/ResizablePanel/ResizablePanel.types';
export * from './molecules/FilterTextField/FilterTextField';
export * from './molecules/ContextMenu/ContextMenu';
export * from './molecules/ContextMenu/ContextMenu.types';
export * from './organisms/HtmlElementPicker/HtmlElementPicker';
export * from './organisms/HtmlElementPicker/HtmlElementPicker.types';
export * from './molecules/Incrementator/Incrementator';
export * from './styles/reactFlowBuilder';
export * from './styles/hiddenNumericArrows';
export * from './molecules/SnackbarActions/SnackbarActions';
export * from './atoms/LightTooltip/LightTooltip';
export * from './atoms/Highlight/Highlight';
export * from './molecules/InformationBox/InformationBox';
