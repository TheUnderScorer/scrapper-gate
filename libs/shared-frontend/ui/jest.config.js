module.exports = {
  displayName: 'shared-frontend-ui',
  preset: '../../../jest.preset.js',
  transform: {
    '^.+\\.[tj]sx?$': 'babel-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
  coverageDirectory: '../../../coverage/libs/shared-frontend/ui',
};
