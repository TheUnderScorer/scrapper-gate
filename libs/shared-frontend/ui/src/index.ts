export * from './atoms/Centered/Centered';
export * from './atoms/Emoji/Emoji';
export * from './molecules/ErrorAlert/ErrorAlert';
