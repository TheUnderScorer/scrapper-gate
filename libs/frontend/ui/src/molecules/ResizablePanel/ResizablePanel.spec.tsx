import { act, render, waitFor } from '@testing-library/react';
import { ResizablePanel } from '@scrapper-gate/frontend/ui';
import userEvent from '@testing-library/user-event';

describe('<ResizablePanel />', () => {
  it('should render without crashing', () => {
    const cmp = render(
      <ResizablePanel minWidth="100px" maxWidth="500px" initialWidth="300px">
        Panel contents
      </ResizablePanel>
    );

    expect(cmp).toMatchSnapshot();
  });

  it('should toggle contents when open/closed', async () => {
    const cmp = render(
      <ResizablePanel initialWidth="300px">Panel contents</ResizablePanel>
    );

    const btn = cmp.container.querySelector('.toggle-panel');

    act(() => {
      userEvent.click(btn);
    });

    const content = cmp.container.querySelector('.resizable-panel-content');
    expect(content.textContent).toEqual('');

    act(() => {
      userEvent.click(btn);
    });

    expect(content.textContent).toEqual('Panel contents');
  });
});
