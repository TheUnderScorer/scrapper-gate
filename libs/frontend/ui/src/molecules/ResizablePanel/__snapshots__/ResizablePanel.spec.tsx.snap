// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResizablePanel /> should render without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="makeStyles-container-2"
        style="position: relative; user-select: auto; width: 300px; height: 100%; max-width: 500px; min-width: 100px; box-sizing: border-box; flex-shrink: 0;"
      >
        <button
          class="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeSmall makeStyles-iconBtn-4 toggle-panel css-8c9wxy-MuiButtonBase-root-MuiFab-root"
          data-mui-internal-clone-element="true"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiFab-label css-110lfi1-MuiFab-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ChevronLeftIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 7.41 14 6l-6 6 6 6 1.41-1.41L10.83 12z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 makeStyles-content-3 resizable-panel-content css-1ps6pg7-MuiPaper-root"
        >
          Panel contents
        </div>
        <div
          class="makeStyles-handle-1"
        >
          <div
            class=""
            style="position: absolute; user-select: none; width: 100%; height: 10px; top: -5px; left: 0px; cursor: row-resize;"
          />
          <div
            class=""
            style="position: absolute; user-select: none; width: 10px; height: 100%; top: 0px; right: -5px; cursor: col-resize;"
          />
          <div
            class=""
            style="position: absolute; user-select: none; width: 100%; height: 10px; bottom: -5px; left: 0px; cursor: row-resize;"
          />
          <div
            class=""
            style="position: absolute; user-select: none; width: 10px; height: 100%; top: 0px; left: -5px; cursor: col-resize;"
          />
          <div
            class=""
            style="position: absolute; user-select: none; width: 20px; height: 20px; right: -10px; top: -10px; cursor: ne-resize;"
          />
          <div
            class=""
            style="position: absolute; user-select: none; width: 20px; height: 20px; right: -10px; bottom: -10px; cursor: se-resize;"
          />
          <div
            class=""
            style="position: absolute; user-select: none; width: 20px; height: 20px; left: -10px; bottom: -10px; cursor: sw-resize;"
          />
          <div
            class=""
            style="position: absolute; user-select: none; width: 20px; height: 20px; left: -10px; top: -10px; cursor: nw-resize;"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="makeStyles-container-2"
      style="position: relative; user-select: auto; width: 300px; height: 100%; max-width: 500px; min-width: 100px; box-sizing: border-box; flex-shrink: 0;"
    >
      <button
        class="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeSmall makeStyles-iconBtn-4 toggle-panel css-8c9wxy-MuiButtonBase-root-MuiFab-root"
        data-mui-internal-clone-element="true"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiFab-label css-110lfi1-MuiFab-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ChevronLeftIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.41 7.41 14 6l-6 6 6 6 1.41-1.41L10.83 12z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 makeStyles-content-3 resizable-panel-content css-1ps6pg7-MuiPaper-root"
      >
        Panel contents
      </div>
      <div
        class="makeStyles-handle-1"
      >
        <div
          class=""
          style="position: absolute; user-select: none; width: 100%; height: 10px; top: -5px; left: 0px; cursor: row-resize;"
        />
        <div
          class=""
          style="position: absolute; user-select: none; width: 10px; height: 100%; top: 0px; right: -5px; cursor: col-resize;"
        />
        <div
          class=""
          style="position: absolute; user-select: none; width: 100%; height: 10px; bottom: -5px; left: 0px; cursor: row-resize;"
        />
        <div
          class=""
          style="position: absolute; user-select: none; width: 10px; height: 100%; top: 0px; left: -5px; cursor: col-resize;"
        />
        <div
          class=""
          style="position: absolute; user-select: none; width: 20px; height: 20px; right: -10px; top: -10px; cursor: ne-resize;"
        />
        <div
          class=""
          style="position: absolute; user-select: none; width: 20px; height: 20px; right: -10px; bottom: -10px; cursor: se-resize;"
        />
        <div
          class=""
          style="position: absolute; user-select: none; width: 20px; height: 20px; left: -10px; bottom: -10px; cursor: sw-resize;"
        />
        <div
          class=""
          style="position: absolute; user-select: none; width: 20px; height: 20px; left: -10px; top: -10px; cursor: nw-resize;"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
